### PR TITLE
Fix warnings in pedantic mode (part1)

### PR DESCRIPTION
**Pull request by dr-pain**

### CODE REVIEW

**Comment by dr-pain** [comment]
make test

**Comment by dr-pain** [comment]
make test

File: include/data_management/data_source/csv_feature_manager.h [hunk 1]
```diff
@@ -451,8 +451,8 @@ class CSVFeatureManager : public StringRowFeatureManagerIface
      *  \param[out] rowBuffer    Pointer to a Buffer View to store the result of parsing
      *  \param[in]  ntRowIndex   Position in the Numeric Table at which to store the result of parsing
      */
-    virtual void parseRowIn(char * rawRowData, size_t rawDataSize, DataSourceDictionary * dictionary,
-                            services::BufferView<DAAL_DATA_TYPE> & rowBuffer, size_t ntRowIndex) DAAL_C11_OVERRIDE
+    virtual void parseRowIn(char * rawRowData, size_t rawDataSize, DataSourceDictionary * /*dictionary*/,
+                            services::BufferView<DAAL_DATA_TYPE> & rowBuffer, size_t /*ntRowIndex*/) DAAL_C11_OVERRIDE
     {
         DAAL_ASSERT(dictionary);
```
Comment (prutskov) [inline include/data_management/data_source/csv_feature_manager.h]: This will not work in debug mode (when we activate _asserts_).

Comment (dr-pain) [reply include/data_management/data_source/csv_feature_manager.h]: fixed. please review

**Comment by dr-pain** [comment]
make test

**Comment by dr-pain** [comment]
PCm - http://intel-ci.intel.com/ea595324-d3f6-f168-9aeb-8cdcd4b723a1?

**Pull request merged**